elephant/statistics.py

```diff
@@ -732,14 +732,33 @@ def instantaneous_rate(spiketrain, sampling_period, kernel='auto',
     ..[1] H. Shimazaki, S. Shinomoto, J Comput Neurosci (2010) 29:171–182.
 
     """
-    # Call function recursively if spiketrain is a list of spike trains
+    # Merge spike trains if list of spike trains given:
     if isinstance(spiketrain, list):
-        rates = []
+        # Check consistency of spike trains
         for st in spiketrain:
-            rate = instantaneous_rate(st, sampling_period=sampling_period, kernel=kernel,
-                                      cutoff=cutoff, t_start=t_start, t_stop=t_start, trim=trim)
-            rates += [rate]
-        return rates
+            if not isinstance(st, SpikeTrain):
```

Comment:
All the consistency check should go into a private function for easier handling.